medusa/server/web/home/handler.py

```diff
@@ -1577,26 +1577,28 @@ def editShow(self, show=None, location=None, allowed_qualities=None, preferred_q
             old_location = os.path.normpath(show_obj._location)
             new_location = os.path.normpath(location)
             if old_location != new_location:
-                logger.log('{old} != {new}'.format(old=old_location, new=new_location), logger.DEBUG)  # pylint: disable=protected-access
-                if not os.path.isdir(location) and not app.CREATE_MISSING_SHOW_DIRS:
-                    errors.append('New location <tt>{location}</tt> does not exist'.format(location=location))
-
-                # don't bother if we're going to update anyway
-                elif not do_update:
-                    # change it
+                logger.log('Changing show location to: {new}'.format(new=new_location), logger.DEBUG)
+                if not os.path.isdir(show_obj._location) and app.CREATE_MISSING_SHOW_DIRS:
+                    logger.log(u"Show directory doesn't exist, creating it", logger.DEBUG)
                     try:
+                        os.mkdir(new_location)
+                        helpers.chmod_as_parent(show_obj._location)
                         show_obj.location = location
+                    except (OSError, IOError) as e:
```

Comment:
You're  doing too much inside the `try`.  Try should wrap the minimum code that could raise the exception.